### PR TITLE
Replace Main Shell Scripts by NodeJS Scripts

### LINKED ISSUE #312: npm start error with Windows
The following error is returned after trying to run `npm start` in the root folder of hospitalrun:

```
> hospitalrun@0.9.0 start Z:\Projects\hospitalrun-frontend
> ./script/server

'.' is not recognized as an internal or external command,
operable program or batch file.

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
npm ERR! node v4.2.6
npm ERR! npm  v2.14.12
npm ERR! code ELIFECYCLE
npm ERR! hospitalrun@0.9.0 start: `./script/server`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the hospitalrun@0.9.0 start script './script/server'.
npm ERR! This is most likely a problem with the hospitalrun package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ./script/server
npm ERR! You can get their info via:
npm ERR!     npm owner ls hospitalrun
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     Z:\Projects\hospitalrun-frontend\npm-debug.log
```

### PR DESCRIPTION
**Pull request by falexsandro**
Fixes #312 

**Changes proposed in this pull request:**
- Replace the main shell scripts with nodejs ones: *build*, *test* and *start*
- New simple scripts using nodejs resources only to replace old shell scripts

cc @HospitalRun/core-maintainers

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,6 +112,7 @@
     "worker-pouch": "git+https://github.com/jkleinsc/worker-pouch.git"
   },
   "dependencies": {
+    "broccoli-funnel": "^2.0.1",
```
Comment (stukalin) [inline package.json]: Was it required?

File: package.json [hunk 2]
```diff
@@ -50,7 +50,7 @@
     "electron-rebuild": "^1.5.7",
     "ember-ajax": "^3.0.0",
     "ember-browserify": "^1.1.12",
-    "ember-cli": "2.17.1",
+    "ember-cli": "^2.18.0",
```
Comment (stukalin) [inline package.json]: Just wondering if you did re-run blueprints, as they recommend https://github.com/ember-cli/ember-cli/releases ?

File: script/build.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env node
```
Comment (stukalin) [inline script/build.js]: Just wondering if we could simply put `ember build` in the `package.json`? The same question goes for `ember test`. And did `-eu` flags have any effect? If yes, was it reflected in your scripts?

File: package.json (hunk 2 above)
Comment (falexsandro) [reply package.json]: It wasn't meant to be part of the commit!
I was learning about semver, and that escaped!
I will rollback that!

File: package.json (hunk 1 above)
Comment (falexsandro) [reply package.json]: That was another mistake!
I imagine that occured too with the tests with semver!
As the other situation, i will rollback that!

File: script/build.js (hunk 3 above)
Comment (falexsandro) [reply script/build.js]: About only calling ember, it's a good question! I'll test later!
About the flags, it just guarantee that script would finalize at first error and variables should be declared before usage! I believe that won't interfere in the ember execution!

Comment (stukalin) [inline script/build.js]: Let's remove those build.js and test.js and merge your PR:)

**Pull request merged**